docs/docs/using.md

```diff
@@ -25,18 +25,27 @@ guidance below.
 </dependency>
 ```
 
-Avoid using [provided] or [optional] scope.
+Avoid using [`provided`] or [`optional`] scope.
 
 #### Gradle
 
+For projects that use the `java-library` plugin, avoid using [`implementation`]
+or (if your users might need the annotations at runtime) [`compileOnlyApi`]
+configurations:
+
 ```groovy
 dependencies {
   api("org.jspecify:jspecify:1.0.0")
 }
 ```
 
-Avoid using [implementation] or (if your users might need the annotations at
-runtime) [compileOnlyApi] configurations.
+Or, for projects that use the `java` plugin and don't publish libraries:
```

Comment:
I don't know enough about the java plugin to know if "and don't publish libraries" is redundant or what.

My bigger fear here is still that, if someone in a hurry sees `implementation("org.jspecify:jspecify:1.0.0")` at the top level, that person will reach for that without considering `api`. My thinking would be that we could start this sentence with something like "Or, if you are using a plugin that does not support `api`, then...."